inspirehep/modules/disambiguation/core/ml/utils.py

```diff
@@ -0,0 +1,330 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction
+
+"""Disambiguation core ML utils."""
+
+from __future__ import absolute_import, division, print_function
+
+from beard.utils import (
+    given_name,
+    given_name_initial,
+    name_initials,
+    normalize_name,
+)
+
+
+def get_author_full_name(s):
+    """Get author full name from the signature.
+
+    Parameters
+    ----------
+    :param s: dict
+        Signature
+
+    Returns
+    -------
+    :returns: string
+        Normalized author name
+    """
+    v = s["author_name"]
```

Comment:
Is it clear that there's always a full name?

Reply:
The name could, for example, be Maqbool S., but in that case, that would be considered the full name. The algorithm works with that.

Reply:
To answer @michamos' original question: yes, because the signature's `author_name` comes from the authors' `full_name` in the record, whose presence is enforced by the schema: https://github.com/inspirehep/inspire-schemas/blob/a542b6efb21a361813ccc80d8a335407672767fd/inspire_schemas/records/hep.yml#L412

Reply:
In particular this mapping is declared here: https://github.com/inspirehep/inspire-next/blob/87a00d2503efb490b16c992671eeb92ac4410938/inspirehep/modules/disambiguation/core/db/readers.py#L150.